### PR TITLE
Re-organize and rename aggregates physical plan

### LINKED ISSUE #2387: Re-organize and rename aggregates physical plan
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

- We currently have a hash-based implementation, `GroupedHashAggregateStream` for aggregate with grouping keys, and a non-hash implementation for aggregate without grouping keys but named `HashAggregateStream`.
- We could further enrich the aggregation method from hash-based to sort-based at runtime when we are run out of memory, as described in https://github.com/apache/arrow-datafusion/issues/1570

**Describe the solution you'd like**
1. Promote `hash_aggregates` to a directory `aggregates`, and re-organize code inside this `aggregates` module.
2. Rename `HashAggregateExec` to `AggregateExec`, since it's not always hashing.

**Describe alternatives you've considered**
Don't move

**Additional context**

### PR DESCRIPTION
**Pull request by yjshen**
# Which issue does this PR close?

Closes #2387.

 # Rationale for this change

- We currently have a hash-based implementation, `GroupedHashAggregateStream` for aggregate with grouping keys, and a non-hash implementation for aggregate without grouping keys but named `HashAggregateStream`.
- We could further enrich the aggregation method from hash-based to sort-based at runtime when we are run out of memory, as described in https://github.com/apache/arrow-datafusion/issues/1570

# What changes are included in this PR?

1. Promote hash_aggregates to a directory aggregates, and re-organize code inside this aggregates module.
2. Rename HashAggregateExec to AggregateExec, since it's not always hashing.

# Are there any user-facing changes?

No.

No.

### CODE REVIEW

File: datafusion/core/src/physical_plan/aggregates/no_grouping.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use futures::{
+    ready,
+    stream::{Stream, StreamExt},
+};
+
+/// stream struct for aggregation without grouping columns
+pub(crate) struct NoGroupingAggregateStream {
```
Comment (yjshen) [inline datafusion/core/src/physical_plan/aggregates/no_grouping.rs:40]: Any suggestions on this single-state aggregation?

File: ballista/rust/core/src/utils.rs [hunk 2]
```diff
@@ -151,7 +151,7 @@ fn build_exec_plan_diagram(
     id: &mut AtomicUsize,
     draw_entity: bool,
 ) -> Result<usize> {
-    let operator_str = if plan.as_any().downcast_ref::<HashAggregateExec>().is_some() {
+    let operator_str = if plan.as_any().downcast_ref::<AggregateExec>().is_some() {
         "HashAggregateExec"
```
Comment (andygrove) [inline ballista/rust/core/src/utils.rs:155]: Should we also change the operator string here to remove Hash ?

**Review by andygrove — APPROVED** [review]
LGTM. I left one question.

File: ballista/rust/core/src/utils.rs (hunk 2 above)
Comment (yjshen) [reply ballista/rust/core/src/utils.rs]: Thanks @andygrove. I've updated all these explanation strings and checked all occurrences of `hash_aggregate` and `HashAggregate`.

File: ballista/rust/core/src/serde/physical_plan/mod.rs [hunk 3]
```diff
@@ -306,19 +306,21 @@ impl AsExecutionPlan for PhysicalPlanNode {
                     Arc::new((&input_schema).try_into()?),
                 )?))
             }
-            PhysicalPlanType::HashAggregate(hash_agg) => {
+            PhysicalPlanType::Aggregate(hash_agg) => {
```
Comment (alamb) [inline ballista/rust/core/src/serde/physical_plan/mod.rs:309]: 👍

File: datafusion/core/src/physical_plan/aggregates/no_grouping.rs (hunk 1 above)
Comment (alamb) [reply datafusion/core/src/physical_plan/aggregates/no_grouping.rs:40]: Maybe just `AggregateStream`?

**Review by alamb — APPROVED** [review]
Looks great -- thanks @yjshen

**Review by andygrove — APPROVED** [review]

**Pull request merged**